### PR TITLE
Improve docstring of x2sys_init and x2sys_cross

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**

This fixes the bullet list in the ``speed`` parameter description in the [x2sys_cross docs](https://www.pygmt.org/dev/api/generated/pygmt.x2sys_cross.html#pygmt.x2sys_cross) in order to suppress this sphinx build warning:

> ``pygmt/pygmt/x2sys.py:docstring of pygmt.x2sys.x2sys_cross:93: WARNING: Bullet list ends without a blank line; unexpected unindent.``

Fixes #

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/x2sys.py [hunk 1]
```diff
@@ -269,8 +269,8 @@ def x2sys_cross(tracks=None, outfile=None, **kwargs):
         - **l** sets lower speed [Default is 0].
         - **u** sets upper speed [Default is Infinity].
         - **h** does not limit the speed but sets a lower speed below which \
-        headings will not be computed (i.e., set to NaN) [Default calculates \
-        headings regardless of speed].
+          headings will not be computed (i.e., set to NaN) [Default \
+          calculates headings regardless of speed].
```
Comment (seisman) [inline pygmt/x2sys.py:273]: I think the newline is unnecessary.
```suggestion
          headings will not be computed (i.e., set to NaN) [Default
          calculates headings regardless of speed].
```

**Comment by seisman** [comment]
I see more tiny issues in the x2sys_cross documentation. Do you want to also fix those tiny issues in this PR?

**Comment by meghanrjones** [comment]
> I see more tiny issues in the x2sys_cross documentation. Do you want to also fix those tiny issues in this PR?

Sure, I can do that.

**Comment by seisman** [comment]
I just converted this PR to draft so save CI resources.

**Comment by meghanrjones** [comment]
@seisman, is there a reason for this module to be in pygmt/, in contrast to the other modules which are in pygmt/src?

**Comment by seisman** [comment]
It should be in `src/`, but we haven't decided if we want `x2sys_init` and `x2sys_cross` in one file `src/x2sys.py`, or two files `src/x2sys_init.py` and `src/x2sys_cross.py`.

See https://github.com/GenericMappingTools/pygmt/pull/832#issuecomment-774356092 and https://github.com/GenericMappingTools/pygmt/issues/807#issuecomment-774784254.

File: pygmt/x2sys.py [hunk 2]
```diff
@@ -93,24 +92,26 @@ def x2sys_init(tag, **kwargs):
...
-        - xyz (same, with one z-column)
-        - geo (for plain ASCII longitude, latitude files)
-        - geoz (same, with one z-column).
+        - **mgd77** (for plain ASCII MGD77 data files)
+        - **mgd77+** (for enhanced MGD77+ netCDF files)
+        - **gmt** (for old mgg supplement binary files)
+        - **xy** (for plain ASCII x, y tables)
+        - **xyz** (same, with one z-column)
+        - **geo** (for plain ASCII longitude, latitude files)
+        - **geoz** (same, with one z-column).
```
Comment (meghanrjones) [inline pygmt/x2sys.py:109]: I have never used this module, so I am not sure whether these should be considered arguments or not (i.e., whether they should be bolded).

**Comment by meghanrjones** [comment]
> > It should be in `src/`, but we haven't decided if we want `x2sys_init` and `x2sys_cross` in one file `src/x2sys.py`, or two files `src/x2sys_init.py` and `src/x2sys_cross.py`.
> > See [#832 (comment)](https://github.com/GenericMappingTools/pygmt/pull/832#issuecomment-774356092) and [#807 (comment)](https://github.com/GenericMappingTools/pygmt/issues/807#issuecomment-774784254).
> 
> Just on this, maybe better to have them in two separate files, since there are other `x2sys_*` modules too which may need to be wrapped in the future. Do we want to do the x2sys move before, during, or after this PR?

I agree about separate files and would prefer making that change during or after this PR to avoid conflicts and extra effort.

**Comment by weiji14** [comment]
> > Just on this, maybe better to have them in two separate files, since there are other `x2sys_*` modules too which may need to be wrapped in the future. Do we want to do the x2sys move before, during, or after this PR?
> 
> I agree about separate files and would prefer making that change during or after this PR to avoid conflicts and extra effort.

Before or after? I can do one before but you'll need to resolve some merge conflicts, or I can open one after this one is merged. I presume this PR is basically done already?

**Comment by meghanrjones** [comment]
> > > Just on this, maybe better to have them in two separate files, since there are other `x2sys_*` modules too which may need to be wrapped in the future. Do we want to do the x2sys move before, during, or after this PR?
> > 
> > 
> > I agree about separate files and would prefer making that change during or after this PR to avoid conflicts and extra effort.
> 
> Before or after? I can do one before but you'll need to resolve some merge conflicts, or I can open one after this one is merged. I presume this PR is basically done already?

I realize now my language was confusing. By 'during' I really meant 'as part of', but in remembering the 'keep PR's small' policy it seems after is best. As far as I can tell this PR is done.

**Review by weiji14 — APPROVED** [review]
Ok, I had a look through the rendered `x2sys_init` and `x2sys_cross` API docs and they seem fine. Though best if @seisman can approve too since he has a better eye on the bold/italics/code-blocks matter.

File: pygmt/x2sys.py [hunk 3]
```diff
@@ -93,24 +92,26 @@ def x2sys_init(tag, **kwargs):
...
 
     suffix : str
         Specifies the file extension (suffix) for these data files. If not
         given we use the format definition file prefix as the suffix (see
-        *fmtfile*).
+        ``fmtfile``).
 
     discontinuity : str
```
Comment (seisman) [inline pygmt/x2sys.py:116]: Line 118. Changing `+ 180` to `+180`.

File: pygmt/x2sys.py [hunk 4]
```diff
@@ -136,25 +137,25 @@ def x2sys_init(tag, **kwargs):
         - **c** - Cartesian userdist or userdist/usertime
```
Comment (seisman) [inline pygmt/x2sys.py:137]: Lines 130-131 are very confusing. I suggest follow the GMT documentation, i.e, **d**|**s***unit*. And *unit* can be **e**, **f** et al.

**Review by seisman — CHANGES_REQUESTED** [review]

File: pygmt/x2sys.py [hunk 5]
```diff
@@ -182,13 +183,13 @@ def x2sys_cross(tracks=None, outfile=None, **kwargs):
...
-    values for all observables. By default, x2sys_cross will look for both
-    external and internal COEs. As an option, you may choose to project all
-    data using one of the map-projections prior to calculating the COE.
+    Determines all intersections between ("external cross-overs") or within
+    ("internal cross-overs") tracks (Cartesian or geographic), and report the
+    time, position, distance along track, heading and speed along each track
+    segment, and the crossover error (COE) and mean values for all observables.
+    By default, :meth:`pygmt.x2sys_cross` will look for both external and
+    internal COEs. As an option, you may choose to project all data using one
+    of the map-projections prior to calculating the COE.
```
Comment (seisman) [inline pygmt/x2sys.py:192]:
```suggestion
    of the map projections prior to calculating the COE.
```

File: pygmt/x2sys.py [hunk 6]
```diff
@@ -221,7 +222,8 @@ def x2sys_cross(tracks=None, outfile=None, **kwargs):
     combitable : str
```
Comment (seisman) [inline pygmt/x2sys.py:222]: Line 210, `\\.gmt` should be `\.gmt` or simply `.gmt`.

File: pygmt/x2sys.py [hunk 7]
```diff
@@ -221,7 +222,8 @@ def x2sys_cross(tracks=None, outfile=None, **kwargs):
...
         Only process the pair-combinations found in the file *combitable*
         [Default process all possible combinations among the specified files].
-        The file *combitable* is created by *x2sys_get*'s -L option
+        The file *combitable* is created by :gmt-docs:`x2sys_get's -L option
+        <supplements/x2sys/x2sys_get.html#l>`.
 
     runtimes : bool or str
         Compute and append the processing run-time for each pair to the
```
Comment (seisman) [inline pygmt/x2sys.py:229]: Line 233: "See the MATLAB function split_file4coes.m that lives in the x2sys supplement source code."

Link to https://github.com/GenericMappingTools/gmt/blob/master/src/x2sys/split_file4coes.m

File: pygmt/x2sys.py [hunk 8]
```diff
@@ -267,10 +269,10 @@ def x2sys_cross(tracks=None, outfile=None, **kwargs):
...
-        - **h** does not limit the speed but sets a lower speed below which \
-        headings will not be computed (i.e., set to NaN) [Default calculates \
-        headings regardless of speed].
+        - **u** sets upper speed [Default is infinity].
+        - **h** does not limit the speed but sets a lower speed below which
+          headings will not be computed (i.e., set to NaN) [Default
+          calculates headings regardless of speed].
 
         For example, you can use ``speed=["l0", "u10", "h5"]`` to set a lower
         speed of 0, upper speed of 10, and disable heading calculations for
```
Comment (seisman) [inline pygmt/x2sys.py:278]: Change `pandas.DataFrame` to <code>:class:\`pandas.DataFrame\`</code>

File: pygmt/x2sys.py (hunk 4 above)
Comment (meghanrjones) [reply pygmt/x2sys.py:136]: Good point, thanks.

File: pygmt/x2sys.py (hunk 8 above)
Comment (weiji14) [reply pygmt/x2sys.py:277]: Why is `pandas.DataFrame` automatically linked under the Parameters section but not under Returns?

Comment (seisman) [reply pygmt/x2sys.py:277]: Good question.

File: pygmt/x2sys.py (hunk 7 above)
Comment (meghanrjones) [reply pygmt/x2sys.py:228]: I linked to the folder rather than the file to keep it under 80 characters.

File: pygmt/x2sys.py [hunk 9]
```diff
@@ -288,11 +290,11 @@ def x2sys_cross(tracks=None, outfile=None, **kwargs):
 
     Returns
     -------
-    crossover_errors : pandas.DataFrame or None
+    crossover_errors : :class:`pandas.DataFrame` or None
         Table containing crossover error information.
         Return type depends on whether the outfile parameter is set:
 
-        - pandas.DataFrame table with (x, y, ..., etc) if outfile is not set
+        - :class:`pandas.DataFrame` with (x, y, ..., etc) if outfile is not set
```
Comment (seisman) [inline pygmt/x2sys.py:297]:
```suggestion
        - :class:`pandas.DataFrame` with (x, y, ..., etc) if **outfile** is not set
```

File: pygmt/x2sys.py [hunk 10]
```diff
@@ -288,11 +290,11 @@ def x2sys_cross(tracks=None, outfile=None, **kwargs):
...
         - None if outfile is set (track output will be stored in outfile)
```
Comment (seisman) [inline pygmt/x2sys.py:298]:
```suggestion
        - None if **outfile** is set (track output will be stored in **outfile**)
```

**Review by seisman — APPROVED** [review]

File: pygmt/x2sys.py (hunk 8 above)
Comment (weiji14) [reply pygmt/x2sys.py:278]: Ok, I see Meghan opened up issue #937 so resolving this.

**Comment by seisman** [comment]
@meghanrjones `/format` and `make format` sometimes don't work well for docstrings. You may have to do it manually.

**Comment by meghanrjones** [comment]
All lines locally appear to be 79 characters or less and my branch is up to date. Should I request a re-run on the style checks?

**Comment by seisman** [comment]
Lines 297 and 298 are too long.
```
pygmt/x2sys.py:297:80: W505 doc line too long (83 > 79 characters)
pygmt/x2sys.py:298:80: W505 doc line too long (81 > 79 characters)
```

**Review by seisman — APPROVED** [review]

**Pull request merged**